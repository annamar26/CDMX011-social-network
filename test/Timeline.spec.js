/**
 * @jest-environment jsdom
 */
import Timeline from '../src/components/Timeline';
import { render } from '../src/utils.js';

describe('Timeline', () => {
  document.body.innerHTML = '<div id="root"></div>';
  const rootDiv = document.getElementById('root');
  it('should render Timeline page', () => {
    render(rootDiv, Timeline);

    expect(rootDiv.innerHTML).toMatchSnapshot();
  });
});
