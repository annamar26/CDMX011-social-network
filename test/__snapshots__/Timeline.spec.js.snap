// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline should render Timeline page 1`] = `
"() =&gt; {
  /* istanbul ignore next */
  cov_21qpngsk2g().f[0]++;
  const container =
  /* istanbul ignore next */
  (cov_21qpngsk2g().s[1]++, document.createElement('div'));
  const html =
  /* istanbul ignore next */
  (cov_21qpngsk2g().s[2]++, \`&lt;nav&gt;
    &lt;div&gt;
    &lt;img class='logo logo1 ' src='./icons&amp;img/logo.png'&gt; &lt;/div&gt;
    &lt;div id='profile' class='icons img-profile'&gt;
    &lt;/div&gt;&lt;/nav&gt;
  
    &lt;div class='timeline'&gt;
   &lt;p&gt;&lt;/p&gt;
   &lt;div id='postCointainer' class='timeline posts-container'&gt;&lt;/div&gt;
   &lt;/div&gt;
   &lt;div id='footer-conteiner'&gt;
  &lt;footer id='footer'&gt;
  &lt;img class='icon' id='createPost' src='./icons&amp;img/createicon.svg'&gt;
  &lt;/footer&gt;&lt;/div&gt;
 
   \`);

  /* istanbul ignore next */
  cov_21qpngsk2g().s[3]++;
  container.innerHTML = html;

  /* istanbul ignore next */
  cov_21qpngsk2g().s[4]++;

  /* istanbul ignore next */
  _firebaseClient.
  /* istanbul ignore next */
  fbFunctions.getPosts();

  /* istanbul ignore next */
  cov_21qpngsk2g().s[5]++;

  /* istanbul ignore next */
  _firebaseClient.
  /* istanbul ignore next */
  fbFunctions.setWelcome();

  /* istanbul ignore next */
  cov_21qpngsk2g().s[6]++;

  /* istanbul ignore next */
  _firebaseClient.
  /* istanbul ignore next */
  fbFunctions.comprobar();

  /* istanbul ignore next */
  cov_21qpngsk2g().s[7]++;
  container.querySelector('#createPost').addEventListener('click', (e) =&gt; {
    /* istanbul ignore next */
    cov_21qpngsk2g().f[1]++;
    cov_21qpngsk2g().s[8]++;
    e.preventDefault();

    /* istanbul ignore next */
    cov_21qpngsk2g().s[9]++;

    /* istanbul ignore next */
    (0,
    /* istanbul ignore next */
    _routes.
    /* istanbul ignore next */
    onNavigate)('/createpost');
  });

  /* istanbul ignore next */
  cov_21qpngsk2g().s[10]++;
  return container;
}"
`;
